### DISCUSSION TITLE
Test on local docker container FluentIcon issues with SSL self signed Certificate

### DISCUSSION THREAD

**Discussion by damianog** [microsoft/fluentui-blazor#437] [Q&A] [answered]
Hi,

I am using Fluent UI in a local docker container for tests, I managed to run the application with my self signed ssl certificate but I have some issues caused by FluentIcon see image below

![image](https://github.com/microsoft/fluentui-blazor/assets/2786682/c3047557-84db-4967-817b-4bec86525060)

Any hint?

**Comment by vnbaaij** [comment, marked as answer]
Have you seen this: https://stackoverflow.com/questions/54402673/how-to-fix-ssl-certificate-problem-self-signed-certificate-in-certificate-chai

**Comment by damianog** [comment]
I managed to run the container, if someone need here my solution:

In Program.cs file 

```csharp
#if DEBUG
            services.AddHttpClient("").ConfigurePrimaryHttpMessageHandler(() => new HttpClientHandler
            {
                ClientCertificateOptions = ClientCertificateOption.Manual,
                ServerCertificateCustomValidationCallback = (httpRequestMessage, cert, cetChain, policyErrors) =>
                {
                    return true;
                }
            });
#else
            services.AddHttpClient();
#endif
```

I use a powershell script to launch the docker compilation, first lines download my local certificate self signed and trusted by dotnet

```powershell
$mypwd = ConvertTo-SecureString -String '1234' -Force -AsPlainText
Get-ChildItem -Path Cert:\LocalMachine\My\ | Where-Object { $_.Subject -like 'CN=localhost' } | Select-Object -First 1 | Export-PfxCertificate -FilePath mylocalhost.pfx -Password $mypwd
Get-ChildItem -Path Cert:\LocalMachine\My\ | Where-Object { $_.Subject -like 'CN=localhost' } | Select-Object -First 1 | Export-Certificate -Type CERT -FilePath mylocalhost.cer
```

inside the classic docker file provided by Microsoft add the installation of the package: ca-certificates
then at the end before the ENTRYPOINT 

```docker
ADD *.pfx /app/
ADD *.cer /usr/local/share/ca-certificates
RUN update-ca-certificates

ENTRYPOINT ["dotnet", "MyApp.dll"]
```

Now you can run the Blazor application in your docker desktop with SSL enabled without issues

```cmd
docker run --rm -it -p 80:80 -p 443:443 -e ASPNETCORE_URLS="https://+;http://+" -e ASPNETCORE_HTTPS_PORT=443 -e ASPNETCORE_Kestrel__Certificates__Default__Password="1234" -e ASPNETCORE_Kestrel__Certificates__Default__Path=mylocalhost.pfx -d --name MyApp MyApp:latest
```
